src/api.js

```diff
@@ -1,15 +1,57 @@
-const fetchArticles = () => {
-    return fetch("https://crowleysnewsapi.onrender.com/api/articles?limit=200")
-      .then((response) => {
-        if (!response.ok) {
-          throw new Error(`HTTP error! Status: ${response.status}`);
-        }
-        return response.json();
-      })
-      .catch((error) => {
-        console.error("Error fetching articles:", error);
-        throw error; // Re-throw the error to handle it in the calling code if needed
-      });
-  };
+const fetchArticles = (articleId) => {
+  let url = "https://crowleysnewsapi.onrender.com/api/articles?limit=200";
+
+  if (articleId) {
+    url = `https://crowleysnewsapi.onrender.com/api/articles/${articleId}`;
+  }
+
+  return fetch(url)
```

Comment:
Axios might save you a lot of faff https://notes.northcoders.com/courses/js-front-end/axios
But to be fair this way you don't need to add a dependancy, so up to you if you want to refactor